viewing_party/party.py

```diff
@@ -0,0 +1,250 @@
+        for movie in user_watched_list:
+            user_movie_title = movie["title"]
+            user_titles_list.append(user_movie_title)
+
+        for friend in friends_watched_list:
+            friend_list = friend["watched"]
+            for movie in friend_list:
+                watched_movie = movie["title"]
+                friends_titles_list.append(watched_movie)
+
+
+        user_titles_set = set(user_titles_list)
+        friends_titles_set = set(friends_titles_list)
+        unique_movies = list(friends_titles_set - user_titles_set)
+
+        friends_unique_movies = []
+        if unique_movies != []:
+            for title in unique_movies:
+                movie = {}
+                movie["title"] = title
+                friends_unique_movies.append(movie)
+        
+            return friends_unique_movies
+        
+        else:
+            return unique_movies
+    else:
+        return friends_titles_list
+
+#WAVE 04
+def get_available_recs(user_data):
+    user_watched_list = user_data["watched"]
+    user_titles_list = []
+    user_sub_list = user_data["subscriptions"]
+    friends_watched_list = user_data["friends"]
+    friends_titles_list = []
+    title_host_list = []
+    rec_host= []
+    rec_movies = []
+    
+    for movie in user_watched_list:
+        user_movie_title = movie["title"]
+        user_titles_list.append(user_movie_title)
+        
+    for friend in friends_watched_list:
+        friend_list = friend["watched"]
+        for movie in friend_list:
+            title_list = movie["title"] 
+            friends_titles_list.append(title_list)
+
+    user_titles_set = set(user_titles_list)
+    friends_titles_set = set(friends_titles_list)
+    user_unwatched = list(friends_titles_set - user_titles_set)
+    for movie in friend_list:
+        for title in user_unwatched:
+            if  title in movie["title"]:
+                title_host_list.append(movie["host"])
+    for host in title_host_list:
+        for sub in user_sub_list:
+            if sub == host:
+                    rec_host.append(host)
+
+    for movie in friend_list:
+        for host in rec_host:
+            if host in movie.values():
+                rec_movies.append(movie)  
+
+    return rec_movies
+
+#WAVE05
+def get_new_rec_by_genre(user_data):
+    user_freq_genre = get_most_watched_genre(user_data)
+    friends_unique_watched = get_friends_unique_watched(user_data)
+    friends_watched_list = user_data["friends"]
+    new_rec_list = []    
+
+    friend_rec = []
+    for watched in friends_unique_watched:
+        friend_watched = watched["title"]
+        friend_rec.append(friend_watched)
+
+        
+    for friend_list in friends_watched_list:
+        movies = friend_list["watched"]
+        for movie in  movies:
+            for title in friend_rec:
+                if title == movie["title"]:
+                    if user_freq_genre == movie["genre"]:
+                        if movie not in new_rec_list:
```

Comment:
Consider refactoring to a single line compound conditional to reduce indentation and enhance readability.